liblwgeom/lwgeom_median.c

```diff
@@ -28,43 +28,50 @@
 #include "lwgeom_log.h"
 
 static void
-calc_distances_3d(const POINT3D* curr, const POINT3D* points, uint32_t npoints, double* distances)
+calc_weighted_distances_3d(const POINT4D* curr, const POINT4D* points, size_t npoints, double* distances)
 {
-	uint32_t i;
+	size_t i;
 	for (i = 0; i < npoints; i++)
 	{
-		distances[i] = distance3d_pt_pt(curr, &points[i]);
+		distances[i] = distance3d_pt_pt((POINT3D*)curr, (POINT3D*)&points[i]) * points[i].m;
```

Comment:
I'd suggest using a weighted distance function that accepts two `POINT4D` pointers, instead of casting to `POINT3D` every time this is called.

Reply:
I'd even prefer to hand-inline the logic as it's a single line :)
Tricky part is to get naming right: the weight is stored just in one of the points, and if you pass two `POINT4D`'s you're passing two of them.